src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3217,8 +3239,18 @@ protected override void VisitArguments(ImmutableArray<BoundExpression> arguments
                         parameterAnnotations,
                         results[i],
                         invokedAsExtensionMethod && i == 0);
+
+                    if (notNullParametersBuilder is object && results[i].RValueType.IsNotNull && (method.Parameters.Last().Equals(parameter) || !expanded))
```

Comment:
>(method.Parameters.Last().Equals(parameter) || !expanded) [](start = 97, length = 57)

Consider returning whether the argument is a value from a `params` array from `GetCorrespondingParameter`.

Reply:
Pending